src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactoryImpl.java

```diff
@@ -31,11 +31,12 @@ public BitbucketCapabilitiesClient getCapabilityClient() {
     }
 
     @Override
-    // TODO: could bitbucketSCMRepo be null?
-    public BitbucketBuildStatusClient getBuildStatusClient(String revisionSha, @Nullable
-            BitbucketSCMRepository bitbucketSCMRepo, BitbucketCICapabilities ciCapabilities) {
+    public BitbucketBuildStatusClient getBuildStatusClient(String revisionSha,
+                                                           @Nullable BitbucketSCMRepository bitbucketSCMRepo,
```

Comment:
Does this need to be nullable? Would there ever be an execution that reaches this place and repo being null? If not, we should remove nullable annotation.

Reply:
Ah yes, I checked this by both mentally following the code path and debugging with a breakpoint, and I very much doubt this will ever be null, because as we discussed in-person, it makes no logical sense for the `BitbucketSCMRepository` to be null when we're posting a Bitbucket-SCM-based build status back to Bitbucket, but I wasn't sure if I could definitely prove so with 100% certainty. I'm happy to treat this as `@Nonnull` and put some safeguards in place _just in case_ it's ever null.

Reply:
Yes, we should mark it as non-null. In general, we should be aggressively removing null as a value. Case in point, it wasted my time and now yours to follow null values.